pkg/pillar/scripts/get-alpine-pkg-source.sh

```diff
@@ -0,0 +1,128 @@
+fi
+
+# Create tmpdir and cd there?
+outdir=/tmp/$$
+mkdir -p $outdir
+cd $outdir || exit
+
+[ -n "$verbose" ] && echo "outdir: $outdir"
+
+INSTALLED=/tmp/installed.$$
+OCPAIRS=/tmp/ocpairs.$$
+truncate --size 0 ${OCPAIRS}
+
+for TAG in ${tags}; do
+    [ -n "$verbose" ] && echo "processing $TAG"
+    rm -f ${INSTALLED}
+    c=$(docker create "${TAG}")
+    docker cp "$c":/lib/apk/db/installed ${INSTALLED}
+    docker rm "$c"
+
+    if [ ! -f ${INSTALLED} ]; then
+	echo "No /lib/apk/db/installed for ${TAG}"
+	continue
+    fi
+
+    # Get pairs of origin and commit. Assumes commit is after origin
+    # otherwise we produce XXX output string
+    # Filter out duplicates
+    awk -F: '
+        /^o:/ { origin=$2 } 
+	/^c:/ { commit=$2; print origin, commit; origin="XXX" }' < ${INSTALLED} | sort -u >> ${OCPAIRS}
+    # shellcheck disable=SC2002
+    [ -n "$verbose" ] && echo "after processing $TAG: $(cat ${OCPAIRS} | wc -l) packages"
+done
+
+# shellcheck disable=SC2002
+[ -n "$verbose" ] && echo "found $(cat ${OCPAIRS} |wc -l) packages $(sort -u ${OCPAIRS} | wc -l) unique"
+
+# shellcheck disable=SC2002
+cat ${OCPAIRS} | sort -u | while read -r line ; do
+    # shellcheck disable=SC2086
+    set -- $line
+    origin=$1
+    commit=$2
+    # Include commit in directory to handle different versions
+    dstdir="${origin}.${commit}"
+    [ -n "$verbose" ] && echo "origin: ${origin} commit: ${commit} dstdir: ${dstdir}"
+    pkgurl="https://git.alpinelinux.org/aports/plain/main/${origin}/APKBUILD?id=${commit}"
+    mkdir "${dstdir}"
+    curl -sSLo "${dstdir}"/APKBUILD "$pkgurl"
+    # XXX is this dangerous? subshell?
+    # shellcheck disable=SC1090
+    source "${dstdir}"/APKBUILD
+    # shellcheck disable=SC2154
+    [ -n "$verbose" ] && echo "source: ${source}"
+    for s in ${source}; do
+	if echo "$s" | grep -sq "https://"; then
+	    [ -n "$verbose" ] && echo "found $s basename $(basename "$s")"
+	    filename="$(basename "$s")"
+	    url="$s"
+	    # Do we need to split on "::"?
+	    if echo "$s" | grep -sq "::"; then
+		# shellcheck disable=SC2001
+		filename="$(echo "$s" | sed 's/^\(.*\)::\(.*$\)/\1/')"
+		# shellcheck disable=SC2001
+		url="$(echo "$s" | sed 's/^\(.*\)::\(.*$\)/\2/')"
+		# XXX try
+		# filename="${s//^\(.*\)::\(.*$\)/\1/}"
+		# url="${s//^\(.*\)::\(.*$\)/\2/}"
+	    fi
+	elif echo "$s" | grep -sq "http://"; then
+	    [ -n "$verbose" ] && echo "found $s basename $(basename "$s")"
+	    filename="$(basename "$s")"
+	    url="$s"
+	    # Do we need to split on "::"?
+	    if echo "$s" | grep -sq "::"; then
+		# shellcheck disable=SC2001
+		filename="$(echo "$s" | sed 's/^\(.*\)::\(.*$\)/\1/')"
+		# shellcheck disable=SC2001
+		url="$(echo "$s" | sed 's/^\(.*\)::\(.*$\)/\2/')"
+		# XXX try
+		# filename="${s//^\(.*\)::\(.*$\)/\1/}"
+		# url="${s//^\(.*\)::\(.*$\)/\2/}"
+	    fi
+	else
+	    [ -n "$verbose" ] && echo "not http*: $s"
+	    filename="$s"
+	    url="${pkgurl}/$s"
```

Comment:
Why do we need to do that? That URL is what works in terms of fetching the content. (Maybe there are other URLs which would also work.)

Reply:
Yes @giggsoff this was a bug resulting in the wrong URLs. Didn't see it because wqe didn't get 404 errors but an error page downloaded instead. Fixed url.